cli.js

```diff
@@ -634,8 +805,22 @@ async function init({ rpc, noteNetId, currency = 'dai', amount = '100' }) {
     TOKEN_AMOUNT = 1e19
     senderAccount = (await web3.eth.getAccounts())[0]
   } else {
-    // Initialize from local node
-    web3 = new Web3(rpc, null, { transactionConfirmationBlocks: 1 })
+    if (torPort) {
+      console.log("Using tor network")
+      web3Options = { agent: { https: new SocksProxyAgent('socks://127.0.0.1:'+torPort) }, timeout: 60000 }
+      // Use forked web3-providers-http from local file to modify user-agent header value which improves privacy.
+      web3 = new Web3(new Web3HttpProvider(rpc, web3Options), null, { transactionConfirmationBlocks: 1 })
```

Comment:
Using forked local providers code instead of the default `web3.providers` only for `tor` connection mode.